table.go

```diff
@@ -54,25 +55,36 @@ func (e ErrCreateSchemaWriter) Error() string {
 }
 
 type TableConfig struct {
-	schema       *dynparquet.Schema
-	rowGroupSize int
+	schema           *dynparquet.Schema
+	rowGroupSize     int
+	blockReaderLimit int
 }
 
 type TableOption func(*TableConfig) error
 
+// WithRowGroupSize sets the size in number of rows for each row group for parquet files. A <= 0 value indicates no limit.
 func WithRowGroupSize(numRows int) TableOption {
 	return func(config *TableConfig) error {
 		config.rowGroupSize = numRows
 		return nil
 	}
 }
 
+// WithBlockReaderLimit sets the limit of go routines that will be used to read persisted block files. A negative number indicates no limit.
+func WithBlockReaderLimit(n int) TableOption {
+	return func(config *TableConfig) error {
+		config.blockReaderLimit = n
+		return nil
+	}
+}
+
 func NewTableConfig(
 	schema *dynparquet.Schema,
 	options ...TableOption,
 ) *TableConfig {
 	t := &TableConfig{
-		schema: schema,
+		schema:           schema,
+		blockReaderLimit: runtime.GOMAXPROCS(0),
```

Comment:
I think for this to work, we need a `_ "go.uber.org/automaxprocs"` import at the top of the file. I would actually be fine merging with `runtime.NumCPU` to avoid any errors (since I think this could currently cause a limit of `0` by mistake) and then adding this intelligent `GOMAXPROCS` things in a separate PR where we go over all concurrency limits in the codebase and expose this through a shared internal package or something.

Reply:
Sounds good

Reply:
> we need a `_ "go.uber.org/automaxprocs"` import at the top of the file.

This should be left to the dependent (e.g. Parca) as this affects the whole application, not frostdb only

Reply:
But what about applications that forget to do this/don't know that they need to do it? IIUC, we need to have some way to be friendly to users that have no idea that they have to set this.

Reply:
It'll default to `numCPU`. If it is imported here, the user cannot opt-out and it could affect other dependencies in ways we cannot control

Reply:
Ok that's true. I guess a value of 0 can't be set either (minimum 1?). In any case, I think it's worth it to encapsulate this logic of choosing default parallelization in a package used by all frostdb components rather than using gomaxprocs directly

Reply:
By "set" i mean set as an env var by an external actor. What I want to avoid is an environment variable causing correctness issues